common/libs/framing/serial_listeners.h

```diff
@@ -0,0 +1,32 @@
+#ifndef __SERIAL_LISTENERS_H
+#define __SERIAL_LISTENERS_H
+
+enum RxError_t {
+  RX_ERROR_UNKNOWN,
+  RX_ERROR_OVR,
+  RX_ERROR_SERIAL_FRAMING,
+  RX_ERROR_TIMEOUT,
+  RX_ERROR_DMA
+};
+
+// An interface that gets called on serial events.
+// NOTE: all callbacks are called from interrupt context!
+class RxListener {
+public:
+  // Called on RX complete
```

Comment:
Suggest renaming and rephrasing:
OnRxCompletedWithoutMarker - called if the requested Rx operation completed successfully without encountering the marker character
OnRxCompletedWithMarker - called if the requested Rx operation completed due to encountering the marker character
OnRxError - called if the requested Rx operation encountered an error

Reply:
UART_DMA does not maintain state if markers are received or not, those events are unrelated from the point of view of UART_DMA.

I would rather keep names are they are because they reflect exactly what happens in UARD_DMA dirver.

Reply:
Okay, could you add more detailed comments though? (e.g. that we receive either exactly N chars, or M<N chars until a marker, or a timeout because we didn't receive at least N chars within the timeout?) Or, in case the more detailed comments about the same concepts are located elsewhere, then point to that place.